.github/workflows/release.yml

```diff
@@ -1,71 +1,91 @@
+    needs: release-please
+    if: ${{ needs.release-please.outputs.release_created }}
 
     steps:
       - uses: actions/checkout@v3
 
       - uses: actions/setup-node@v3
         with:
           node-version: '18'
+          registry-url: 'https://registry.npmjs.org'
+          cache: 'npm'
 
-      - name: Restore node_modules
-        uses: actions/cache@v3
-        with:
-          path: |
-            node_modules
-          key: npm-${{ runner.os }}-18.x-${{ hashFiles('package-lock.json') }}
-
-      - name: Install dependencies
-        run: npm ci
-
-      - name: Build
-        run: npm run build
-
-      - name: Create zip
-        uses: vimtor/action-zip@v1
-        with:
-          # Be sure to keep this in sync with the `files` array in `package.json`
-          files: build/ scripts/ generated-types/ bundles/.empty_directory db/.empty_directory schemas/ AUTHORS package-lock.json package.json
-          recursive: true
-          dest: nodecg.zip
+      - run: npm ci
+      - run: npm run build
 
-      - name: Semantic Release
-        run: npx semantic-release
+      - name: Publish to github release
+        run: |
+          FILE_NAME=$(npm pack)
+          gh release upload ${{ needs.release-please.outputs.release_name }} $FILE_NAME
         env:
           GH_TOKEN: ${{ github.token }}
-          GITHUB_TOKEN: ${{ github.token }}
-          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+      - name: Publish to npm
+        run: npm publish --access public
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
 
       - name: Publish types
         run: |
           npm run generate-types-package
           cd generated-types
-          npm publish --public
+          npm publish --access public
         env:
-          NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
+
+  docker-image:
+    runs-on: ubuntu-latest
+    needs: release-please
+    if: ${{ needs.release-please.outputs.release_created }}
+
+    steps:
+      - uses: actions/checkout@v3
 
       - name: Log in to the Container registry
         uses: docker/login-action@v2
         with:
-          registry: ${{ env.REGISTRY }}
+          registry: ghcr.io
           username: ${{ github.actor }}
-          password: ${{ secrets.GITHUB_TOKEN }}
+          password: ${{ github.token }}
 
       - name: Extract metadata (tags, labels) for Docker
         id: meta
         uses: docker/metadata-action@v4
         with:
-          images: ${{ env.REGISTRY }}/${{ env.IMAGE_NAME }}
+          images: ghcr.io/${{ github.repository }}
+          tags: |
+            type=ref,event=branch
+            type=semver,pattern={{major}}.{{minor}}.{{patch}},value=${{ needs.release-please.outputs.release_name }}
+            type=semver,pattern={{major}}.{{minor}},value=${{ needs.release-please.outputs.release_name }}
+            type=semver,pattern={{major}},value=${{ needs.release-please.outputs.release_name }}
```

Comment:
this custom tags definitions are required to push images with tags like `1.2.3` `1.2` or `1` when the action was triggered by a commit